numpy/core/src/multiarray/number.c

```diff
@@ -565,7 +565,7 @@ array_negative(PyArrayObject *m1)
 static PyObject *
 array_absolute(PyArrayObject *m1)
 {
-    if (can_elide_temp_unary(m1)) {
+    if (can_elide_temp_unary(m1) && !PyArray_ISCOMPLEX(m1)) {
```

Comment:
hm right, it shouldn't elide object arrays at all it only does that for unary not bianry probably I forgot to update both places.